ci.md

```diff
@@ -0,0 +1,53 @@
+# CI pipeline overview
+
+This is an overview of the CI pipeline within starchart
+
+## Concurrency group
+
+The `Concurrency` group is used to ensure only one instance of the CI job is running per branch. For example, if 2 commits are pushed in close session to one another, the first commits CI run will be cancelled in favor of the second run.
+
+This is done through defining a concurrency group with a unique identifier: `group: ${{ github.workflow }}-${{ github.ref }}`, in which only 1 CI run can occupy at a time. The newer CI run will always take presence over older runs.
+
+## Jobs
+
+All CI jobs are run concurrently, and utilize caching where possible.
+
+In every job, it will attempt to fetch `node_modules` from a cache, instead of installing it fresh
+
+```yaml
+- name: Restore node_modules cache
+  id: node_modules-cache-restore
+  uses: actions/cache/restore@v3
+  with:
+    # path to node_modules to cache
+    path: ./node_modules
+    # cache module with hash of package-lock.json
+    key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+```
+
+The hash key is the os name followed by the package-lock hash(`Linux-node-63fe6c0ea79bc1812337982f05a442b9a9ee4df56c872a7f4dad8df440079be5`). This is so when app dependencies are updated, and package-lock is regenerated. It will download and cache the newer `node_modules` instead of restoring the older version from cache.
```

Comment:
```suggestion
The hash key is the os name followed by the package-lock hash(`Linux-node-63fe6c0ea79bc1812337982f05a442b9a9ee4df56c872a7f4dad8df440079be5`). This is so when app dependencies are updated and package-lock is regenerated, it will download and cache the newer `node_modules` instead of restoring the older version from cache.
```

This period seems unintentional. Not sure if this is the best fix though